pkg/discovery/discovery_test.go

```diff
@@ -156,83 +147,74 @@ var _ = Describe("DiscoveryNG", func() {
 				})
 				go s.Start()
 				s.WaitUntilReady(timeout)
-				pb.PublishExternalEvent(playerOneIsReady, ClientIncomingEventsTopic)
-				pb.PublishExternalEvent(playerOneIsReady, ClientIncomingEventsTopic)
+
+				for i := 0; i < playerCount; i++ {
+					// ToDo: Isn't it an error if the same player sends its ReadyMessage multiple times?
+					//    Or rather, should we in such a case really go to PlayersReady?
```

Comment:
Could you have another look at this ToDo statement?
We were not sure if the way the test was written before and the way its also working now is supposed to work at all, or whether it should be a protocol error instead?

Before: 2 Players, State: [Waiting for PlayerReady events].
Player 1 sends Ready event twice
Result: State switch to PlayersReady -> Even though Player2 never sent its own ready event!

Now: Same but with N players
N Players in Game, State [Waiting for PlayerReady events]
Player 1 sends Ready event N times
Result: State switch to PlayersReady -> Even though Players 2 to N never sent their own ready event!

Reply:
As of my understanding (Disclaimer: not my code), this test is not testing actual discovery client functionality as it uses `FakeDClient` (see line 59). When you look at the implementation you see that a `PlayersReady` state event is triggered after a single `PlayerReady` event comes in: 

https://github.com/carbynestack/ephemeral/blob/12c134bdde64033c9faee7360dcd06cf07ddcaa8/pkg/discovery/fakes.go#L33-L51

Had no chance yet to dig into it and fully understand the rationale. Is it important to address this in this PR. I'd suggest to file an issue to track and investigate this further. Agree?